ts/components/ChooserListSearch.tsx

```diff
@@ -0,0 +1,174 @@
+    paddingLeft: customVariables.contentPadding,
+    paddingTop: customVariables.contentPadding,
+    textAlign: "center"
+  },
+  emptySubtitle: {
+    fontSize: customVariables.fontSizeBase,
+    paddingRight: customVariables.contentPadding,
+    paddingLeft: customVariables.contentPadding,
+    paddingBottom: customVariables.contentPadding,
+    textAlign: "center"
+  }
+});
+/**
+ * A component that renders a list of object that match a search text.
+ */
+class ChooserListSearch<T> extends React.PureComponent<Props<T>, State<T>> {
+  constructor(props: Props<T>) {
+    super(props);
+    this.state = {
+      potFilteredListStates: pot.noneLoading
+    };
+  }
+
+  /**
+   * Filter only the object that match the searchText.
+   */
+  private generateListStateMatchingSearchTextAsync(
+    listState: ReadonlyArray<T>,
+    searchText: string
+  ): Promise<ReadonlyArray<T>> {
+    const { onSearchItemContainsText } = this.props;
+    // Don't apply the filter if searchText is empty
+    if (searchText.length === 0) {
+      return Promise.resolve(listState);
+    }
+    return new Promise(resolve => {
+      const result = listState.filter(
+        _ =>
+          onSearchItemContainsText
+            ? onSearchItemContainsText(_, searchText)
+            : false
+      );
+      resolve(result);
+    });
+  }
+
+  private renderResultEmptyComponent() {
+    const { noSearchResultsSourceIcon, noSearchResultsSubtitle } = this.props;
+    return (
+      <View style={styles.emptyContentWrapper}>
+        <View spacer={true} />
+
+        <Text style={styles.emptyTitle} bold={true}>
+          {I18n.t("global.search.noResultsTitle")}
+        </Text>
+
+        {noSearchResultsSubtitle && (
+          <Text style={styles.emptySubtitle}>{noSearchResultsSubtitle}</Text>
+        )}
+
+        <Image
+          source={
+            noSearchResultsSourceIcon
+              ? noSearchResultsSourceIcon
+              : require("../../img/search/beer-mug.png")
+          }
+        />
+        <View spacer={true} />
+      </View>
+    );
+  }
+
+  public async componentDidMount() {
+    const { listState, searchText } = this.props;
+    const { potFilteredListStates } = this.state;
+
+    // Set filtering status
+    this.setState({
+      potFilteredListStates: pot.toLoading(potFilteredListStates)
+    });
+
+    // Start filtering list
+    const filteredServiceSectionsStates = await this.generateListStateMatchingSearchTextAsync(
+      listState,
+      searchText
+    );
+
+    // Unset filtering status
+    this.setState({
+      potFilteredListStates: pot.some(filteredServiceSectionsStates)
+    });
+  }
+
+  public async componentDidUpdate(prevProps: Props<T>) {
+    const { listState: prevListState, searchText: prevSearchText } = prevProps;
+    const { listState, searchText } = this.props;
+    const { potFilteredListStates } = this.state;
+
+    if (listState !== prevListState || searchText !== prevSearchText) {
```

Comment:
What do you about using a [debounced](https://github.com/teamdigitale/io-app/blob/712963804d8b5d97005bf94591524bfdd96c708b/ts/components/search/SearchButton.tsx#L100) approach to avoid executing a search for each typed char?

Reply:
Ok it's a good idea. I proceed

Reply:
@Undermaken Done in commit 7d0f7ce.